server/keyshare/tasks/tasks.go

```diff
@@ -126,29 +176,40 @@ func (t *taskHandler) sendExpiryEmails(ctx context.Context, id int64, username,
 	return nil
 }
 
-// Mark old unused accounts for deletion, and inform their owners.
+// expireAccounts marks old unused accounts for deletion and informs their owners.
+// When email revalidation is enabled, email addresses which where marked for revalidation are skipped
+// because these will be processed separately inside revalidateEmails.
 func (t *taskHandler) expireAccounts(ctx context.Context) {
 	// Disable this task when email server is not given
 	if t.conf.EmailServer == "" {
 		t.conf.Logger.Warning("Expiring accounts is disabled, as no email server is configured")
 		return
 	}
 
-	// Iterate over users we have not seen in ExpiryDelay days, and which have a registered email.
-	// We ignore (and thus keep alive) accounts without email addresses, as we can't inform their owners.
-	// (Note that for such accounts we store no email addresses, i.e. no personal data whatsoever.)
-	// We do this for only 10 users at a time to prevent us from sending out lots of emails
-	// simultaneously, which could lead to our email server being flagged as sending spam.
-	// The users excluded by this limit will get their email next time this task is executed.
-	err := t.db.QueryIterateContext(ctx, `
+	query := `
 		SELECT id, username, language
 		FROM irma.users
 		WHERE last_seen < $1 AND (
 			SELECT count(*)
 			FROM irma.emails
 			WHERE irma.users.id = irma.emails.user_id
+			{{revalidate}}
 		) > 0 AND delete_on IS NULL
-		LIMIT 10`,
+		LIMIT 10`
```

Comment:
This means that if the user has two email addresses registered, and one of those addresses is being revalidated, then we continue to expire its account... I'd say that we wait until we know what the result of the revalidation will be. This is only a few days.

Reply:
Applied the following logic:
- If one of multiple email addresses is valid, the user will receive an email about expiry. In that case I do think it is logical if we continue expiring the account, so that's what happens.
- When none of multiple addresses is valid, then account expiry will not happen until revalidation has done it's thing.

Reply:
I've a bit of a dilemma. If you registered multiple email addresses, would you either want to get a message on all your email addresses or would you think its okay if you only receive it on a subset of the email address. In case it was caused by an explicit action of the user, then the latter one seems more logical to me. Otherwise, if one email address has issues you cannot do the action. But if it concerns something automatic, like account expiry, I tend to prefer the former one, but it's only a subtle difference. What do you think?

Reply:
Updated the tasks implementation to match the discussed and agreed solution: only if all email addresses are valid the expiry mail is sent. This means if a user has one or more invalid addresses, these will first go through the revalidation process.